examples/util/BaseRunner.js

```diff
@@ -103,283 +132,93 @@ class BaseRunner {
+   * @param options: {object | undefined}
+   * An object for options has backend and prefer info.
+   * options = {
+   *   backend: backend, // 'WASM' | 'WebGL' | 'WebML'
+   *   prefer: prefer, // 'fast' | 'sustained' | 'low' | 'ultra_low'
+   * }
+   */
+  compileModel = async (options) => {
+    // Override by inherited
+  };
+
+  /**
+   * This method is to do inference with input src (HTML [<img> | <video> | <audio>] element).
+   * The id of <img> element is fixed as 'feedElement' in index.html.
+   * The id of <video> element or <audio> element is fixed as 'feedMediaElement' in index.html.
+   * @param src: An object for HTML [<img> | <video> | <audio>] element.
+   * @param options: An object to get input tensor. Details:
+   * options = {
+   *   // inputSize was configed in modelZoo.js, inputSize = [h, w, c] or [1, size] for audio example.
+   *   inputSize: inputSize,
+   *   // preOptions was also configed in modelZoo.js,
+   *   // preOptions= {} or {mean: [number, number, number, number],std: [number, number, number, number]}
+   *   preOptions: preOptions,
+   *   imageChannels: 4,
+   *   drawOptions: { // optional, drawOptions is used for CanvasRenderingContext2D.drawImage() method.
+   *     sx: sx, // the x-axis coordinate of the top left corner of sub-retangle of the source image
+   *     sy: sy, // the y-axis coordinate of the top left corner of sub-retangle of the source image
+   *     sWidth: sWidth, // the width of the sub-retangle of the source image
+   *     sHeight: sHeight, // the height of the sub-retangle of the source image
+   *     dWidth: dWidth, // the width to draw the image in the detination canvas
+   *     dHeight: dWidth, // the height to draw the image in the detination canvas
+   *   },
+   *   scaledFlag: true, // optional, need scaled the width and height of element to get need inputTensor
+   * };
+   */
   run = async (src, options) => {
-    let status = 'ERROR';
-
-    if (src.tagName === 'AUDIO') {
-      await getTensorArrayByAudio(src, this._inputTensor, options);
-    } else {
-      getTensorArray(src, this._inputTensor, options);
-    }
-
-    const start = performance.now();
-    status = await this._model.compute(this._inputTensor, this._outputTensor);
-    const delta = performance.now() - start;
-    this._setInferenceTime(delta);
-    console.log(`Computed Status: [${status}]`);
-    console.log(`Compute Time: [${delta} ms]`);
-    return status;
-  };
-
-  getRequiredOps = () => {
-    return this._modelRequiredOps;
+    // Override by inherited
   };
 
-  getSubgraphsSummary = () => {
-    return this._subgraphsSummary;
-  };
-
-  getDeQuantizeParams = () => {
-    return this._deQuantizeParams;
-  };
-
-  _updateOutput = (output) => {
+  /**
+   * This method is to output inference tensor for post processing by example side.
+   * @param output: An object for output which will be updated with output tensor info by this method.
+   */
+  _getOutputTensor = (output) => {
```

Comment:
change it to return a tensor object (typed array?). see following comment for `getOutput`.